src/snowflake/cli/plugins/snowpark/models.py

```diff
@@ -65,6 +67,73 @@ def to_requirement_with_files(self):
         return RequirementWithFiles(self.requirement, self.files)
 
 
+@dataclass
+class RequirementWithWheelAndDeps:
+    """A dataclass to hold a requirement and corresponding .whl file."""
+
+    requirement: Requirement
+    wheel_path: Path | None
+    dependencies: List[str]
+
+    def extract_files(self, destination: Path) -> None:
+        if self.wheel_path is not None:
+            with zipfile.ZipFile(self.wheel_path, "r") as whl:
+                whl.extractall(destination)
+
+    def namelist(self) -> List[str]:
+        if self.wheel_path is None:
+            return []
+        with zipfile.ZipFile(self.wheel_path, "r") as whl:
+            return whl.namelist()
+
+
+@dataclass
+class WheelMetadata:
+    """A dataclass to hold metadata from .whl file.
+    [name] is the name of the package standarized accroding to
+    https://peps.python.org/pep-0491/#escaping-and-unicode
+    """
+
+    name: str
+    wheel_path: Path
+    dependencies: List[str]
+
+    @classmethod
+    def from_wheel(cls, wheel_path: Path):
```

Comment:
What if there's no wheel distribution?

Reply:
`pip wheel` builds and packs a wheel from what is available - this was the easiest approach to have consistent artifacts and avoid "install" at the same time